ui/ui-components/pages/export/[id]/edit.tsx

```diff
@@ -27,9 +33,47 @@ export const getServerSideProps = withServerErrorHandler(async (ctx) => {
 });
 
 const Page: NextPageWithInferredProps<typeof getServerSideProps> = ({
-  _export,
   groups,
+  ...props
 }) => {
+  const { client } = useApi();
+  const [_export, setExport] = useState(props._export);
+  const [loading, setLoading] = useState(false);
+
+  const retryExport = useCallback(async () => {
```

Comment:
A lot of this logic seems to be repeated in two files. I wonder if we could encapsulate part, if not all of this logic, maybe in a hook?

Reply:
I think the best option may be to extract the meat of the function and just export that. Otherwise you'll end up passing in all the hook setters, etc. I'll move it, lemme know what you think!